docs/docs/adrs/adr-019-fault-resolutions.md

```diff
@@ -0,0 +1,131 @@
+
+In what follows, we describe the implementation of a fault resolution mechanism for any intersubjective fault.
+ Note that in the first iteration, it is only incorrect executions that are defined as a fault and are therefore dealt with by the mechanism (see [Incorrect Executions](https://forum.cosmos.network/t/preventing-intersubjective-faults-in-ics/14103#incorrect-execution-fault-definition-5) in "Preventing Intersubjective faults in ICS").
+
+
+## Decision
+
+The proposed solution introduces a new `consumer-fault-resolution` governance proposal type to the `provider` module, which allows validators to be penalised for committing faults on an Opt-in consumer chain.
+
+If such a proposal passes, the proposal handler tombstones all the validators listed in the proposal and slashes them by a per-consumer chain predefined
+ amount or the default value used for double-sign infractions.
+
+The proposal has the following fields:
+
+- **Consumer Chain**: The consumer chain ID that the fault was related to.
+- **Validators**: The list of all the validators to be slashed.
+- **Evidence**: A free text form.
+- **Fault Type**: The fault definition type.
+- **Description**: This field is automatically generated by aggregating the fault definition corresponding to the *Fault Type* and the *Evidence* fields.
+
+ Each fault type is mapped to a fault definition that precisely describes an intersubjective fault, such as an incorrect execution, and explains why it qualifies as a slashable fault. Refer to the [fault definitions section](https://forum.cosmos.network/t/preventing-intersubjective-faults-in-ics/14103#fault-definitions-4) in "Preventing Intersubjective faults in ICS" for more details. Note that the text of each fault definition is stored as a string constant in the provider code.
+
+
+In addition, to prevent spamming, users must pay a default fee of `100ATOM` to submit a fault resolution to the provider.
+ This amount is stored in a new `consumer-fault-resolution-fee` parameter of the `provider` module.
+
+### Validations
+
+The submission of a fault resolution succeeds only if all of the following conditions are met:
+
+- the consumer chain is an Opt-in chain
+- all listed validators were opted-in to the consumer chain in the past unbonding-period
+- the `100ATOM` fee is provided
+
+### States
+
+Additional states are added to the `provider` modules:
+
+* The timestamps that record when validators opts in or opts out of a Opt-in consumer chain. 
+ Note that these timestamps can be pruned after an unbonding period elapses following a validator's opts-out.
+
+```golang
+    ConsumerValidatorSubscriptionTimestampPrefix | len(consumerID) | consumerID | valAddr | ProtocolBuffer(ConsumerValSubscriptionTimestamp)
+```
+
+```protobuf
+    messsage {
+        // timestamp recording the last time a validator opted in to the consumer chain
+        google.protobuf.Timestamp join_time = 1;
```

Comment:
Maybe add a sentence on why `join_time` is needed. The way I see it, we could simply get by using the `leave_time`. If the validator is currently opted in, then the `leave_time` is irrelevant. If the validator is not currently opted in, then we can see if they left before unbonding period or not.